crates/noirc_evaluator/src/ssa_refactor/acir_gen/mod.rs

```diff
@@ -1,15 +1,17 @@
 //! This file holds the pass to convert from Noir's SSA IR to ACIR.
+
+pub(crate) use self::generated_acir::GeneratedAcir;
+
 use super::ssa_gen::Ssa;
 
+mod generated_acir;
+
 /// Context struct for the acir generation pass.
 /// May be similar to the Evaluator struct in the current SSA IR.
 struct Context {}
 
-/// The output of the Acir-gen pass
-pub(crate) struct Acir {}
-
 impl Ssa {
-    pub(crate) fn into_acir(self) -> Acir {
+    pub(crate) fn into_acir(self, _main_param_array_lengths: &[usize]) -> GeneratedAcir {
```

Comment:
Passing `main_param_array_lengths` seems like it should be an implementation detail of `into_acir` rather than the responsibility of the caller. It is fine for this milestone but I think for the future we should consider passing the whole abi instead.

Reply:
Yeah this should be fixed in M1 @joss-aztec